[IMP] inventory: UPS config (#8975)
odoo/documentation: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/ups_credentials.rst

===============
UPS integration
===============

UPS is a shipping carrier service that integrates with Odoo to coordinate shipping to all regions.

> Once integrated, users can create shipping methods that estimates shipping costs and :doc:`generate

Reviewer: plural/singular mismatch I think?
```suggestion
Once integrated, users can create shipping methods that estimate shipping costs and :doc:`generate
```

labels <../advanced_operations_shipping/labels>`.

.. seealso::
   :doc:`third_party_shipper`

To set up the UPS shipping connector in Odoo, complete these steps:

#. Create a UPS account and get the credentials
#. Configure company information in UPS
#. Set up shipping method in Odoo

> An Access Key is a 16 character alphanumeric code that allows access to

Reviewer: This is the only time an access key is mentioned in this article...do we actually need it?
If so can we add more context around it, as it seems out of place after the overview list right before we get into the config.

the UPS Developer Kit API Development and Production servers.

UPS account setup
=================

To get started, go to the `UPS website <https://www.ups.com>`_ and click the :guilabel:`Log In`
button in the top-right corner to log in or create a UPS account.

After logging in, click the profile icon in the top-right corner, and select :guilabel:`Accounts and

> Payments` from the drop-down menu.

Reviewer: **Payment** (singular), as it's shown in the screenshot
```suggestion
Payment` from the drop-down menu.
```

.. image:: ups_credentials/accounts-payment.png
   :align: center
   :alt: Show how to navigate to the "Accounts and Payment" page from the home screen.

> On the :guilabel:`Accounts and Payments` page, two accounts must be configured: an Odoo shipment

Reviewer: ```suggestion
On the :guilabel:`Accounts and Payment Options` page, two accounts must be configured: an Odoo shipment
```
![ups-accounts-and-payment-options](https://github.com/odoo/documentation/assets/5496809/a738b1ca-ccf8-4434-aed1-ed705556e839)

account and a payment card.

Shipping account
----------------

To add an Odoo shipment account, select :guilabel:`Add New Account` from the :guilabel:`Add a
Payment Method` drop-down menu, and click :guilabel:`Add`.

.. image:: ups_credentials/new-account.png
   :align: center
   :alt: Display "Add an account" option from the drop-down menu.

> Next, complete the forms to configure company details and address for setting up the shipping
> account.

Reviewer: ```suggestion
On the next screen, labeled :guilabel:`Open a Shipping Account`, complete the forms to configure the
shipping account type (e.g. :guilabel:`Business`) and if any regulated items will be shipped. Then
finish the remaining three steps in the wizard to :guilabel:`Add Addresses`, :guilabel:`Verify
Identity`, and :guilabel:`Explore Discounts`, with the last option being optional.

When complete, submit the application on the last page of the wizard to finish setting up the
shipping account.
```

![ups-open-a-shipping-account](https://github.com/odoo/documentation/assets/5496809/dc09b3a4-b656-4899-b8e0-63df0a90d151)

it, navigate to :menuselection:`Profile --> Accounts and Payment` and refer to the shipping
account's :guilabel:`Number` field.

.. image:: ups_credentials/account-number.png
   :align: center
   :alt: Show Account "Number" field for the shipping account.

Payment card
------------

Navigate back to the :guilabel:`Accounts and Payments` page and select the :guilabel:`Add Payment
Card` option from the :guilabel:`Add a Payment Method` drop-down menu. Then, complete the form to
add the credit card information.

.. image:: ups_credentials/payment-card.png
   :align: center
   :alt: Show "Add Payment Card" option from the drop-down.

UPS developer account setup
===========================

Next, log into the `UPS developer account <http://developer.ups.com/>`_ to generate the developer
key. To begin, click the profile icon in the top-right corner, and choose the :guilabel:`Apps`
option from the drop-down menu.

.. image:: ups_credentials/apps.png
   :align: center
   :alt: Display the "Apps" drop-down option after clicking the profile picture icon.

Add app
-------

Then, click the :guilabel:`Add Apps` button to begin filling out the form. In the :guilabel:`I need

> API credentials because` field, select :guilabel:`I want to integrate UPS technology into my

Reviewer: Consider adding the asterisk since it's part of the gui
We can do this in RST by adding an escape key (backslash) in front of it so it's not rendered as italics.
```suggestion
API credentials because \*` field, select :guilabel:`I want to integrate UPS technology into my
```

business`.

> In :guilabel:`Choose an account to associate with these credentials`, select :guilabel:`Add existing
> account` and select the :ref:`account number <inventory/shipping_receiving/ups-account-number>`
> linked to the UPS account created in the previous step.

Reviewer: specificity
also, clarification, do we *select* the account (is a pre-filled option) or do we have to add it ourselves (i.e. via text input)?
```suggestion
Under the next label, :guilabel:`Choose an account to associate with these credentials. \*`, select
:guilabel:`Add existing account` from the drop-down menu in the corresponding field, and then select
the :ref:`account number <inventory/shipping_receiving/ups-account-number>` linked to the UPS
account created in the previous step.
```

Author: Since I wasn't able to test this part, I don't know whether we select the account number or type it :sweat_smile:

.. image:: ups_credentials/developer-account-setup.png
   :align: center
   :alt: Show form to fill in the UPS account number.

Click :guilabel:`Next`, and proceed to the :guilabel:`Add App` form, and fill out the fields:

- :guilabel:`App Name`: Type the name to identify the app by.

> - :guilabel:`Callback URL`: Type the URL of the Odoo database, in the format: `https://` + database
>   name + `.odoo.com`. Do **not** include `www.`

Reviewer: optional feedback, this is how I would do it since some readers might misinterpret the `+` signs (as in, they will literally add them into the URL field 😄 )

```suggestion
- :guilabel:`Callback URL`: Type the URL of the Odoo database, in the format:
  `https://databaseName.odoo.com`. Do **not** include `www` in the URL.
```

Author: great way of showing the URL!

   :align: center
   :alt: Show "Add Apps" form, where the app details are configured.

Finally, click :guilabel:`Save` and accept UPS's terms and conditons. With the new app created,
select the app from the :guilabel:`My Apps` section to view the UPS credentials.

.. image:: ups_credentials/my-apps.png
   :align: center
   :alt: Show newly created app in the "My Apps" section.

In the :guilabel:`Credentials` section, copy the :guilabel:`Client ID` and :guilabel:`Client Secret`
key.

.. _inventory/shipping_receiving/ups-client-id:

Client ID
---------

To obtain the account's :guilabel:`Client ID`, click the profile and select :menuselection:`Apps`
from the drop-down menu, and select the Odoo shipping account.

In the :guilabel:`Credentials` section, copy the :guilabel:`Client ID`.

.. image:: ups_credentials/credentials.png
   :align: center
   :alt: Display the "Client ID" and "Client Secret" key.

.. _inventory/shipping_receiving/ups-client-secret:

Client Secret
-------------

To obtain the account's :guilabel:`Client Secret` key, click the profile and select
:menuselection:`Apps` from the drop-down menu, and select the Odoo shipping account.

> In the :guilabel:`Credentials` section, copy the :guilabel:`Client Secret`.

Reviewer: The instructions here seem redundant and actually aren't as clear or concise as line 138, while we're in the **My Apps > App > Credentials** section.

I think you can remove this whole block, but keep the screenshot since that's relevant to the instruction on line 138.

<third_party_shipper>` documentation.

In the :guilabel:`UPS Configuration` tab, complete the following fields:

- :guilabel:`UPS Account Number`: (*required*) Get the :ref:`account number
  <inventory/shipping_receiving/ups-account-number>` from the UPS portal.
- :guilabel:`UPS Client ID`: (*required*) Get the :ref:`Client ID
  <inventory/shipping_receiving/ups-client-id>` from the UPS developer website.
- :guilabel:`UPS Client Secret`: (*required*) Get the :ref:`Client Secret
  <inventory/shipping_receiving/ups-client-secret>` key from the UPS developer website.
- :guilabel:`UPS Service Type`: Select from the drop-down menu the type of shipping service.
- :guilabel:`UPS Package Type`: (*required*) Select from the drop-down menu the :doc:`package type
  <../../product_management/product_tracking/package>` that is supported for the shipping service.
- :guilabel:`Package Weight Unit`: The unit of measure for the package weight.
- :guilabel:`Package Size Unit`: The unit of measure for the package dimensions.
- :guilabel:`Label Format`: Choose the label format shipping labels: :guilabel:`PDF`,
  :guilabel:`ZPL`, :guilabel:`EPL`, or :guilabel:`SPL`.

> .. image:: ups_credentials/ups-configuration.png

Reviewer: clever way to show what's required without having to add red boxes!
